agent/host_agent_test.go

```diff
@@ -188,4 +192,45 @@ var _ = Describe("Agent", func() {
 			Consistently(session.Err, "10s").ShouldNot(gbytes.Say(byoHost.Name))
 		})
 	})
+
+	Context("When host agent is executed with --version flag", func() {
+		var (
+			tmpHostAgentBinary string
+		)
+		BeforeEach(func() {
+			date, err := exec.Command("date").Output()
+			Expect(err).NotTo(HaveOccurred())
+			version.BuildDate = string(date)
+			version.Version = "1.2.3"
+			ldflags := fmt.Sprintf("-X 'github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/agent/version.Version=%s'"+
+				" -X 'github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/agent/version.BuildDate=%s'", version.Version, version.BuildDate)
+			fmt.Fprint(GinkgoWriter, ldflags)
+			tmpHostAgentBinary, err = gexec.Build("github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/agent", "-ldflags", ldflags)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			version.BuildDate = ""
+			version.Version = ""
+			tmpHostAgentBinary = ""
+		})
+
+		It("Shows the appropriate version of the agent", func() {
+			expectedStruct := version.Info{
+				Major:     "1",
+				Minor:     "2",
+				Patch:     "3",
+				BuildDate: version.BuildDate,
+				GoVersion: runtime.Version(),
+				Compiler:  runtime.Compiler,
+				Platform:  fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
+			}
+			expected := fmt.Sprintf("byoh-hostagent version: %#v\n", expectedStruct)
+			out, err := exec.Command(tmpHostAgentBinary, "--version").Output()
+			output := string(out)
+			fmt.Fprintf(GinkgoWriter, "version.Version %s", version.Version)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
Can you move this a couple lines up where it gets the `err` value?

Reply:
fixed